libraries/AP_Scripting/applets/ScriptingSwitcher/Script_Controller.md

```diff
@@ -1,15 +1,21 @@
+
+# Optional Mission and Parameter switching
+
+It is also possible to switch and load a different mission or set of parameters to accompany the new set of scripts at the same time. Simply include a copy of Mission_Controller.lua and/or Param_Controller.lua in each subdirectory that you desire the mission or some parameters changed along with the new scripts to be executed. 
 
 # Setup and Operation
 
-The user must setup an RCx_OPTION to function 302 to determine which of the three scripts subdirectories will be used: LOW:scripts/1, MIDDLE:scripts/2, or HIGH:scripts/3. If no RC has been established during ground start, it will behave as if subdirectory 1 is selected. Changing this switch position either prior to ground start or after, will remove ALL the *.lua files in the scripts directory (except itself), and copy any *.lua files (only) from the desginated subdirectory, and then issue a scripting restart. 
+The user must setup an RCx_OPTION to function 302 to determine which of the three scripts subdirectories will be used: LOW:scripts/1, MIDDLE:scripts/2, or HIGH:scripts/3. If no RC has been established during ground start, it will behave as if subdirectory 1 is selected. Changing this switch position either prior to ground start or after, will remove ALL the *.lua files in the scripts directory (except itself), and copy any *.lua files (only) from the designated subdirectory, and then issue a scripting restart. 
+
+If the mission or parameter script options are used for a subdirectory the mission file to be loaded at the same time that subdirectory's scripts are loaded should be named "mission.txt" and/or "params.param".
 
 # CAUTIONS
 
 If parameters are being created and used be sure that different scripts do not use the same PARAMETER KEY to create their param tables, unless you desire to use the same parameter set in two different scripts. Otherwise, param values will be corrupt when you change scripts.
-Be sure to have unique parameter names for different parameter sets. Names are only visible and used by Ground Control Stations. While indentically named params in different scripts will operate fine, you will not be able to distinguish them in the GCS displays!
+Be sure to have unique parameter names for different parameter sets. Names are only visible and used by Ground Control Stations. While identically named params in different scripts will operate fine, you will not be able to distinguish them in the GCS displays!
 
 Any .lua script in the scripts directory will be ERASED! when this script is run. In order to keep another .lua script running when using this script, place it in every subdirectory so that it will be copied over when changing directories.
```

Comment:
suggestion: place a copy in **every** subdirectory

Reply:
you may not want or have to do that...only the subs that have a mission to load...not needed in other subs that do not since when you switch to them you do not want a mission change

Reply:
Have you tested that? As I read it if you had another unrelated lua script you wanted to always be there - something that has nothing to do with the mission - if you don't put a copy of it in each sub directory, it will be deleted if you switch to a directory where there is no copy of that script.

Reply:
absolutely